lib/setopt.c

```diff
@@ -930,6 +930,9 @@ CURLcode Curl_vsetopt(struct Curl_easy *data, CURLoption option, va_list param)
     data->set.http09_allowed = arg ? TRUE : FALSE;
 #endif
     break;
+  case CURLOPT_FORM_ESCAPE_AS_MIME:
+    data->set.form_escape_as_mime = va_arg(param, long) != 0;
```

Comment:
The manpage says the value has to be 1 so I think it can just as well check for that?

Reply:
This is  handled as other bool options. The man page has been cut'n pasted from `CURLOPT_VERBOSE`!
I prefer to change the man page.

Reply:
Obsolete since this is a bitmask option now. manpage rewritten.

Reply:
Just a note on the previous design: we once used "non-zero argument for..." (which is why some source code still does that check) but over time we've learned that it is much better to specifically say and check for exact numbers (like `1`), since that makes it non-ambiguous and saves other values for future use. Should a need occur in the future when something else comes up and we want to introduce a variation.

Reply:
Understood and noted :-)